test/d_do_test.d

```diff
@@ -792,14 +809,23 @@ int tryMain(string[] args)
             if (testArgs.disabled)
                 return Result.return0;
 
+            if (envData.autoUpdate)
```

Comment:
Looks like this is missing curly braces.

Reply:
No that's a short-hand syntax. Unfortunately assignments in `if` are quite limited, so `if (envData.autoUpdate && auto ce = cast(CompareException) e)` doesn't work.

Reply:
I understand why you're doing that, but such syntax is always going to raise an eyebrow because it looks like a bug. I'll approve it because it is in the test harness and I'll defer to you on style in it - in the compiler/phobos I'd also ask to change it.